### PR TITLE
Add bottom bar navigation component

### PR DESCRIPTION
**Pull request by ricardo7martins**
https://github.com/theolm/instagram-clone/assets/46760836/422339cc-3ba6-4910-93da-d1e4d9281157

https://github.com/theolm/instagram-clone/assets/46760836/a674cae7-cca6-405c-a937-3c6857804e9e

### CODE REVIEW

File: api-client/build.gradle.kts [hunk 1]
```diff
@@ -5,6 +5,9 @@ plugins {
 android {
     namespace = "com.theolm.temp"
 }
+dependencies {
```
Comment (gabrielbmoro) [inline api-client/build.gradle.kts:8]: Do we need this dependency here?

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceAround
+    ) {
+        IconButton(onClick = {
+            selectSingleTab(homeTabState, tabList)
+        }) {
+            Icon(
+                painterResource(if (homeTabState.value) MR.images.home_selected else MR.images.home_unselected),
+                modifier = iconModifier,
+                contentDescription = "Home Menu"
```
Comment (gabrielbmoro) [inline composeApp/src/commonMain/kotlin/common/BottomBar.kt:64]: Can we move this to be the string resources?

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        }
+        IconButton(onClick = {
+            selectSingleTab(searchTabState, tabList)
+        }) {
+            Icon(
+                painterResource(
+                    if (searchTabState.value) MR.images.search_selected else MR.images.search_unselected
+                ),
+                modifier = iconModifier,
+                contentDescription = "Search Menu"
```
Comment (gabrielbmoro) [inline composeApp/src/commonMain/kotlin/common/BottomBar.kt:75]: Can we move this to be the string resources?

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            )
+        }
+        IconButton(onClick = {
+        }) {
+            Icon(
+                painterResource(MR.images.add_new),
+                modifier = iconModifier,
+                contentDescription = "Add Menu"
```
Comment (gabrielbmoro) [inline composeApp/src/commonMain/kotlin/common/BottomBar.kt:83]: Can we move this to be the string resources?

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            )
+        }
+        IconButton(onClick = {
+            selectSingleTab(favoriteTabState, tabList)
+        }) {
+            Icon(
+                painterResource(if (favoriteTabState.value) MR.images.favorite_selected else MR.images.favorite_unselected),
+                modifier = iconModifier,
+                contentDescription = "Favorite Icon"
```
Comment (gabrielbmoro) [inline composeApp/src/commonMain/kotlin/common/BottomBar.kt:92]: Can we move this to be the string resources?

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            )
+        }
+        IconButton(modifier = Modifier
+            .border(1.dp, profileBorderColor.value, RoundedCornerShape(100)).size(27.dp), onClick = {
+            selectSingleTab(profileTabState, tabList)
+        }) {
+            profileBorderColor.value = if (profileTabState.value) Color.Black else Color.Transparent
+            /*TODO ("Replace with dynamic obtained image from user")*/
```
Comment (gabrielbmoro) [inline composeApp/src/commonMain/kotlin/common/BottomBar.kt:100]: Can we create an issue for that?

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.unit.dp
+import dev.icerock.moko.resources.compose.painterResource
+import instaclone.resources.MR
+
+@Composable
+fun BottomBar() {
```
Comment (gabrielbmoro) [inline composeApp/src/commonMain/kotlin/common/BottomBar.kt:30]: I think you can make this component dynamic, something like:

```kotlin
data class BottomBarItem(
        val isSelected: Boolean, 
        val iconRes: ImageResource,
        val contentDescriptionRes: StringResource,
...
)

@Composable
fun BottomBar(bottomBarItems: List<BottomBarItem>, modifier: Modifier = Modifier) {
     ...
}
```

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt (hunk 3 above)
Comment (ricardo7martins) [reply composeApp/src/commonMain/kotlin/common/BottomBar.kt]: Done!

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt (hunk 2 above)
Comment (ricardo7martins) [reply composeApp/src/commonMain/kotlin/common/BottomBar.kt]: Done!

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt (hunk 6 above)
Comment (ricardo7martins) [reply composeApp/src/commonMain/kotlin/common/BottomBar.kt]: Hopefully, but I'm guessing API integration of any sort is far down the line

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt (hunk 7 above)
Comment (ricardo7martins) [reply composeApp/src/commonMain/kotlin/common/BottomBar.kt]: I reorganize the component like so: e6d412d7c92e3237c2683d6a0bff9907fe0b777f
Is this enough?

File: composeApp/src/commonMain/kotlin/common/BottomBar.kt (hunk 4 above)
Comment (ricardo7martins) [reply composeApp/src/commonMain/kotlin/common/BottomBar.kt]: Done!

File: api-client/build.gradle.kts (hunk 1 above)
Comment (ricardo7martins) [reply api-client/build.gradle.kts]: It's removal seems to have no impact. So i'll take it out

**Pull request merged**